### ISSUE TITLE
Integration of a YunoHost registration form

### ISSUE THREAD

**Issue by collector-ynh** [YunoHost/issues#1992]
Hello,

For more than **8 years** the YunoHost community has been requesting a registration form external, because it is an elementary need for administrators in order to make their communities benefit from the SSO connection of YunoHost and all these applications.

Here is again what the community is asking for: https://github.com/YunoHost/issues/issues/1677

It is amazing to see that such an elementary feature has still not been implemented in YunoHost despite the various relaunches. Here is a non-running list of several discussions regarding the integration of a registration form to YunoHost :

https://forum.yunohost.org/t/how-to-use-the-api/889
https://forum.yunohost.org/t/registration-page-for-yunohost/18058
https://forum.yunohost.org/t/registration-app/2561
https://forum.yunohost.org/t/api-yunohost-pour-creation-de-comptes-utilisateurs/17373
...

It is it's a real shame not to make YunoHost's SSO users benefit because of the impossibility of registering for the latter.

**Comment by alexAubin** [comment]
Waow

You really have either a communication problem, or a misunderstanding of what this project is, or probably both.

I already tried to point it out on the chat but you really don't seem to understand the issue with your attitude and your tone. And there are other people finding it difficult to deal with your attitude. Thanksfuly the vast majority of the YunoHost is usually much more friendly, but from time to time we end up with people not understanding the deal with this kind of project.

So I'm going to explain this one last time before we start taking actions. It's honestly kinda disturbing to explain this, I really feel like explaining why 2+2=4. YunoHost is and has always been a volunteer project, and it's free both as in free beer and as in freedom. This has several implications:
- Things are done on a volunteering basis. Things are released "when they are deemed ready". There are no hard deadline, there are no "forecast" on when features (or even bugged) may be integrated
- The project doesn't owe you anything. You may **not** come on the chat, the forum or this issue tracker to  "I am a user and I *demand* that the community packages whichever app I want", and neither to rant that "OMG I am a user and I am outraged that this feature is not still implemented !!1!1". Which doesn't mean that you're not allowed to add apps to the wishlist, or create/bump a thread on the issue tracer to signify that some feature/fix is important. See, we're talking about the tone of communication as if the project "owes you something" and volunteers are at your orders. (cf also next item)
- The most valuable resource of the project is volunteer time and energy. Being pushy and arrogant is pissing off volunteers, because what is often a time-consuming hobby turns into a non-paid job, which is naturally not very sustainable. Hence, people wasting the time and energy of volunteers are harmful to the project, and if the toxic attitude continues, we shall put an end to it.

Now more precisely about your post:

>For more than **8 years**

Yeah, cool, at this point I only red 5 words in your post and already understand that it's gonna be a good ride

>the YunoHost community has been requesting <insert feature>

c.f. previous discussion, the YunoHost project is not a company obeying the community's orders. Volunteers work on what they want at any given time and aren't tied by any contract. (Also, again stating the obvious, but anybody in the world can volunteer to work on anything, it's not like we're a closed circle or whatever). Volunteers coordinate each other according to what is deemed important in the huge todolist, and to some extend what's "fun" to work on.

>a registration form external, because it is an elementary need for administrators in order to make their communities benefit from the SSO connection of YunoHost and all these applications.
>
>Here is again what the community is asking for: https://github.com/YunoHost/issues/issues/1677

So you are well aware that a ticket discussing this feature, but are still opening one for the same feature. I'm not sure what you're trying to achieve by doing so. Will opening 10 tickets all named "Self-registration form" make it happen 10 times faster ?

As explained before, if anything, your post primarily effect is to discourage volunteers because nobody wants to deal with ungrateful users demanding more as if they're kings.

>It is amazing to see that such an elementary feature has still not been implemented in YunoHost despite the various relaunches. 

c.f previous discussion, YunoHost is not a company at your orders. The project doesn't owe anything to anybody.

>blah blah blah more links blah blah this is a shame

What is truly amazing here is that you don't realize that, more than anybody, volunteers of the project know by heart the list of "important long-term topics", which - can you believe it? - cannot be counted with just fingers and toes. This situation is reminiscent to a recent situation where a person much like you repeatedly reminded us that there was this thing called Bullseye around the corner, to which our reaction was pretty much this : 

![youdontsay](https://user-images.githubusercontent.com/4533074/159612196-0bffeb02-f6c7-46ab-8c00-a2e8861563c4.jpg)

Like, yeah, wow, people want self-registration form ? Damn, we didn't know. Thanks for telling us about it. We never ever heard of it in our X years of contributing to this project countless of hours per week. Now that we know about it, it's just gonna be ... 15 minutes of code and *voila*!

**Comment by alexAubin** [comment]
Duplicate of https://github.com/YunoHost/issues/issues/1677
